lambda/create_jira.py

```diff
@@ -25,39 +27,48 @@
 JIRA_USERNAME = os.environ["JIRA_USERNAME"]
 
 secrets_manager = boto3.client('secretsmanager')
-
 secret = secrets_manager.get_secret_value(SecretId=JIRA_API_TOKEN_SECRET_ARN)
-logger.error(secret)
-
 JIRA_API_TOKEN = secret['SecretString']
 
 
 # establish connection to jira
-options = {"server": JIRA_URL}
-
 jira = JIRA(
     basic_auth=(JIRA_USERNAME, JIRA_API_TOKEN),
-    options={
-        'server': JIRA_URL
-    }
-)
+    options={'server': JIRA_URL})
+
+print("Connected to Jira: {}".format(jira.server_info()))
 
-logger.info("Connected to Jira: {}".format(jira.server_info()))
+
+def parse_message(message):
+
+    if type(message) is str:
+        try:
+          message = json.loads(message)
+        except json.JSONDecodeError as err:
+          logger.error(f'JSON decode error: {err}')
+
+    return message
 
 
 def lambda_handler(event, context):
+    """ receives events from SNS, see https://docs.aws.amazon.com/lambda/latest/dg/with-sns.html """
+
+    print("Event received: {}".format(event))
+
+    subject = event['Records'][0]['Sns']['Subject']
+    message = parse_message(event['Records'][0]['Sns']['Message'])
+    region = event['Records'][0]['Sns']['TopicArn'].split(":")[3]
 
-    logger.debug("Event received: {}".format(event))
+    description = "\n".join([ f'{k}: {v}' for k, v in message.items()])
 
     issue_fields = {
         'project': JIRA_PROJECT,
-        'summary': 'Alert - {}'.format(event['id']),
-        'description': event,
-        'issuetype': {'name': ISSUE_TYPE}
-    }
+        'summary': 'Alert - {}'.format(subject),
+        'description': description,
+        'issuetype': {'name': ISSUE_TYPE}}
 
     issue = jira.create_issue(fields=issue_fields)
     jira.add_comment(
-        issue.key, 'Alert triggered by {} AWS CloudWatch integration'.format(INTEGRATION_NAME))
+        issue.key, 'Alert triggered by {} aws cloudwatch integration'.format(INTEGRATION_NAME))
```

Comment:
y?

Reply:
oops

Reply:
I'm not nearly as good at Vim as I should be and sometimes I mash buttons 🤷

Reply:
I always mash buttons